pkgs/applications/virtualization/OVMF/default.nix

```diff
@@ -1,19 +1,23 @@
-{ stdenv, lib, edk2, nasm, iasl, seabios, openssl, secureBoot ? false }:
+{ stdenv, lib, buildPackages, edk2, nasm, iasl, seabios, openssl, secureBoot ? false }:
 
 let
-
   projectDscPath = if stdenv.isi686 then
     "OvmfPkg/OvmfPkgIa32.dsc"
   else if stdenv.isx86_64 then
     "OvmfPkg/OvmfPkgX64.dsc"
-  else if stdenv.isAarch64 then
+  else if stdenv.isAarch64 || stdenv.isAarch32 then
     "ArmVirtPkg/ArmVirtQemu.dsc"
   else
     throw "Unsupported architecture";
 
   version = (builtins.parseDrvName edk2.name).version;
 
-  src = edk2.src;
+  inherit (edk2) src;
+  hostArch = buildPackages.edk2.targetArch;
+
+  buildFlags = if stdenv.isAarch64 then ""
+    else if seabios == null then ''${lib.optionalString secureBoot "-DSECURE_BOOT_ENABLE=TRUE"}''
+    else ''-D CSM_ENABLE -D FD_SIZE_2MB ${lib.optionalString secureBoot "-DSECURE_BOOT_ENABLE=TRUE"}'';
 in
 
 stdenv.mkDerivation (edk2.setup projectDscPath {
```

Comment:
I see below (cannot comment there) `buildInputs = [nasm iasl]` which is definitely wrong. Try to cross compile this derivation too---building firmware should always be cross compilation anyways.

Reply:
Yeah, GitHub may have eaten a previous reply I wrote. It worked mostly due to luck. I never cross-compiled to x86_64 from a non-x86_64 platform, making that `nasm` work. As for `iasl`, it was in the `nativeBuildInputs` from `edk2.setup` masking this issue.

For every time I push, the code has been tested to cross-compile and run for aarch64, and natively build on x86_64.